docs/sec-blueprint/07-Hardening.md

```diff
@@ -0,0 +1,597 @@
+|  CONFIG\_MODULE\_SIG\_FORCE=y         |3.7+            |
+|  CONFIG\_PACKET\_DIAG=n               |3.7+            |
+|  CONFIG\_FW\_LOADER\_USER\_HELPER=n   |3.9+            |
+|  CONFIG\_CC\_STACKPROTECTOR=y         |3.11+ (MIPS)    |
+|  CONFIG\_USELIB=n                     |3.15+           |
+|  BPF\_JIT=n                           |3.16+ (MIPS)    |
+|  CONFIG\_DEVMEM=n                     |4.0+            |
+
+### Build with Stack Protection
+
+  Similar to the stack protector used for ELF programs in user-space,
+  the kernel can protect its internal stacks as well. This configuration
+  for the MIPS architecture is supported in Linux 3.11 and greater and
+  thus should only be enabled for such versions. This configuration also
+  requires building the kernel with the gcc compiler 4.2 or greater.
+
+```
+  CONFIG_CC_STACKPROTECTOR=y
+```
+
+### Disable access to /dev/mem
+
+  The /dev/mem file in Linux systems is directly mapped to physical
+  memory. This can be disastrous if an attacker gains root access, as
+  the attacker would have direct access to physical memory through this
+  convenient device file. It may not always be possible to disable such
+  file, as some applications might need such support. In that case then
+  this device file should be available only for authenticated
+  applications. This configuration is supported in Linux 4.0 and greater
+  and thus should only be disabled for such versions.
+
+```
+  CONFIG_DEVMEM=n
+```
+
+### Disable cross-memory attach
+
+  Disable the process\_vm\_\*v syscalls which allow one process to
+  peek/poke the virtual memory of another. This configuration is
+  supported in Linux 3.5 and greater and thus should only be disabled
+  for such versions.
+```
+  CROSS_MEMORY_ATTACH=n
+```
+
+### Disable core dumps
+
+  Core dumps provide lot of debug information for hackers. So disabling
+  core dumps is recommended in production builds. This configuration is
+  supported in Linux 3.7 and greater and thus should only be disabled
+  for such versions.
+
+```
+  CONFIG_COREDUMP=n
+```
+
+### Disable Legacy Linux Support
+
+  There are some Kernel Configs which are present only to support legacy
+  binaries. See also section 2.2.2.18 for disabling support for legacy
+  binary formats. The uselib system call, in particular, has no valid
+  use in any libc6 or uclibc system in recent times. This configuration
+  is supported in Linux 3.15 and greater and thus should only be
+  disabled for such versions.
+
+```
+  CONFIG_USELIB=n
+```
+
+### Disable firmware auto-loading user mode helper
+
+  The firmware auto loading helper, which is a utility executed by the
+  kernel on hotplug events requiring firmware, needs to be set setuid.
+  As a result of this, the helper utility is an attractive target for
+  attackers with control of physical ports on the device. Disabling this
+  configuration is supported in Linux 3.9 and greater.
+
+```
+  CONFIG_FW_LOADER_USER_HELPER=n
```

Comment:
True, but we need to check which drivers need firmware and where it should be allowed.